src/components/Glp/GlpSwap.js

```diff
@@ -649,6 +652,33 @@ export default function GlpSwap(props) {
     }
   };
 
+  const [pageTracked, setPageTracked] = useState(false);
+
+  const dataElements = [chainId, isBuying, pageTracked, swapTokenAddress, history.location.hash];
+  const elementsLoaded = Object.values(dataElements).every((element) => element !== undefined);
```

Comment:
JS freaky, could you also just use dataElements.every?

Reply:
cannot because that always returns true, i need to test for the values of the json objects

Reply:
e.g. 
```
{
  chainId: 421611,
  tableView: undefined,
}
```
^ still returns true

Reply:
but if dataElements is an array the values are just the items in the array?